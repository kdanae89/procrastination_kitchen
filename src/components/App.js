import React from 'react';
import spoonacular from '../api/spoonacular';
import Header from './Header/Header';
import IngredientsSearch from './IngredientsSearch/IngredientsSearch';
import RecipeList from './RecipeList/RecipeList';
import RecipeDetail from './RecipeDetail/RecipeDetail';
import './App.scss';

class App extends React.Component {
  state = {
    recipes: [],
    selectedRecipe: null
  };

  onSearchSubmit = async (ingredients) => {
    const response = await spoonacular.get('/recipes/findByIngredients', {
        params: {
          "ranking":"2",
          "ignorePantry":"false",
          "ingredients":ingredients
        }
      });

      this.setState({ recipes: response.data });
  };

  onRecipeSelect = (recipe) => {
    this.setState({ selectedRecipe: recipe });
  };

  render () {
    const isSideList = this.state.selectedRecipe ? true : false;

    return (
      <div className="container">
        <Header />
        <IngredientsSearch onSubmit={this.onSearchSubmit} />
        <div className="result-wrapper">
          {this.state.selectedRecipe && (
            <RecipeDetail recipe={this.state.selectedRecipe} />
          )}
          <RecipeList
            isSideList={isSideList}
            onRecipeSelect={this.onRecipeSelect}
            recipes={this.state.recipes}
          />
        </div>
      </div>
    );
  }
}

export default App;
