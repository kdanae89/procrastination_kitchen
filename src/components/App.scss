.container {
  width: 98vw;
}

.result-wrapper {
  display: flex;
  height: 90vh;
  justify-content: space-between;
  margin-top: 2em;
}

.dimmed.dimmable {
  display: flex !important;
  align-items: center;
  justify-content: center;
}

.ui.page.active.modals {
  position: absolute;
}
