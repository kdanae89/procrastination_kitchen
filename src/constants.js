export const APP_TITLE = 'Procrastination Kitchen';
export const LOCAL_URI = 'http://localhost:1337';
