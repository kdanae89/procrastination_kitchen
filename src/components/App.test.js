import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import axios from 'axios';

import App, { onSearchSubmit, Header, IngredientsSearch, RecipeList } from './App';
import { APP_TITLE } from '../constants.js';

const mockStore = configureStore([]);

describe('Connected App should render', () => {
  let store;
  let context;
  let app;

  beforeEach(() => {
    store = mockStore({
      state: {
        selectedRecipe: null
      }
    });

    context = mount(
      <Provider store={store}>
        <App />
      </Provider>
    );

    app = context.find(App);
  });

  it('should render App', () => {
    context = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );

    expect(context.toJSON()).toMatchSnapshot();
  });
  it('should contain Header', () => {
    expect(app.find({children: APP_TITLE}).length).toEqual(1);
  });
  it('should contain IngredientSearch', () => {
    expect(app.find({children: "What's in your pantry?"}).length).toEqual(1);
  });
  it('should not contain RecipeList', () => {
    expect(app.find({'data-testid': 'recipe-list'}).length).not.toEqual(1);
  });
  //it should call mock fn
  //it should change state after onSubmit
  //it should render recipeList after onFormSubmit
});

describe('On Search', () => {
  let store;
  let context;
  let app;

  beforeEach(() => {
    store = mockStore({
      state: {
        selectedRecipe: null
      }
    });


    context = mount(
      <Provider store={store}>
        <App />
      </Provider>
    );




    app = context.find(App);
  });

  jest.spyOn(App.prototype, "onSearchSubmit");

  const promise = new Promise((resolve, reject) =>
    setTimeout(
      () =>
        resolve({
          data: {
            recipes: [
              {
                title: "pickle soup",
                missedIngredientCount: 1
              },
              {
                title: "pickle tacos",
                missedIngredientCount: 3
              }
            ]
          }
        }),
      100
    )
  );

  it('should call the mock onSearchSubmit', () => {

  })
})
