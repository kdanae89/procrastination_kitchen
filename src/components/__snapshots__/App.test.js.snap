// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected App should render should render App 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <div
      className="trans"
    >
      <div
        className="navigation"
      >
        <i
          className="material-icons"
        >
          menu
        </i>
      </div>
      <h1
        className="title"
      >
        Procrastination Kitchen
      </h1>
      <div
        className="user-profile"
      >
        <i
          className="material-icons"
        >
          account_box
        </i>
      </div>
    </div>
  </div>
  <div
    data-testid="search"
  >
    <form
      className="search-input"
      data-testid="search-input"
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <label>
          What's in your pantry?
        </label>
        <input
          data-testid="input"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </form>
  </div>
  <div
    className="result-wrapper"
  />
</div>
`;
